Microsoft.Toolkit.Uwp.UI/Converters/ByteArrayToImageConverter.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using Windows.Storage.Streams;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Microsoft.Toolkit.Uwp.UI.Converters
+{
+    /// <summary>
+    /// This class converts a byte[] value into BitmapImage.
+    /// If the value is null or not a byte array, it will return null
+    /// Otherwise the logic will try to convert it to BitmapImage
+    /// So it can be used with the Image Control.
+    /// </summary>
+    public class ByteArrayToImageConverter : IValueConverter
+    {
+        /// <summary>
+        /// Convert a byte[] value to BitmapImage.
+        /// </summary>
+        /// <param name="value">An image as byte array.</param>
+        /// <param name="targetType">The type of the target property, as a type reference.</param>
+        /// <param name="parameter">Optional parameter. Not used.</param>
+        /// <param name="language">The language of the conversion. Not used</param>
+        /// <returns>BitmapImage</returns>
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            if (value is byte[] byteArray)
+            {
+                using InMemoryRandomAccessStream stream = new();
```

Comment:
Don't think we need this complexity here. We should just be able to pass the byte[] to a MemoryStream constructor and pass that into SetSource?

effectively `image.SetSource(new MemoryStream(bytes).AsRandomAccessStream())`

Reply:
the change cause the app to hang forever but the converter has a bigger problem for now https://github.com/CommunityToolkit/WindowsCommunityToolkit/pull/4330#discussion_r739532038 because of the `BitmapImage` type